psulib_config.rb

```diff
@@ -92,17 +97,70 @@
   246abcdefgnp
   247abcdefgnp
 ].join(':'))
-
 to_field 'title_added_entry_tsim', extract_marc(%w[
   700gklmnoprst
   710fgklmnopqrst
   711fgklnpst
   730abcdefgklmnopqrst
   740anp
 ].join(':'))
+to_field 'title_related_tsim', extract_marc(%w[
+  505t
+  700lktmnoprs
+  710lktmnoprs
+  711lktmnoprs
+  730adfgklmnoprst
+  740anp
+  760st
+  762st
+  765st
+  767st
+  770st
+  772st
+  773st
+  774st
+  775st
+  776st
+  777st
+  780st
+  785st
+  786st
+  787st
+  790lktmnoprs
+  791lktmnoprs
+  792lktmnoprs
+  793adflktmnoprs
+  796lktmnoprs
+  797lktmnoprs
+  798lktmnoprs
+  799alktmnoprs
+].join(':'), trim_punctuation: true) do |record, accumulator|
+  accumulator.each { |value| value.chomp!(' --') } unless record.fields('505').empty?
+end
 
-to_field 'title_series_tsim', extract_marc('440anpv:490av')
+## Title Display Fields
+to_field 'title_latin_display_ssm', extract_marc('245abcfgknps', alternate_script: false, trim_punctuation: true)
```

Comment:
Maybe this should be just `title_latin`?

Reply:
this field is displayed in the item level main content when the vernacular is the main title so it is part of the index 🤔

Reply:
I like your approach, let me test it on my local.

Reply:
I'm not sure if one way is more efficient, I would guess your way is slightly faster, fwiw. Again, both ways are totally fine IMO.